src/Microsoft.ML.OnnxTransformer/OnnxTransform.cs

```diff
@@ -383,10 +383,12 @@ private protected override void SaveModel(ModelSaveContext ctx)
         /// </summary>
         private static IEnumerable<int> AdjustDimensions(OnnxShape shape)
         {
-            if (shape.Count > 0)
-            {
-                return shape.Select(x => (x <= 0) ? 1 : x);
-            }
+            if (shape.Count > 0 && shape[0] < 0)
+                shape[0] = 1;
+
+            if (shape.Count > 1)
+                return shape.Select(x => (x <= 0) ? 0 : x);
+
             return new[] { 1 };
```

Comment:
>return new[] { 1 }; [](http://example.com/codeflow?start=12&length=19)

can this be a static array if no one write to it?

Reply:
Its possible there is more than one onnx model in a pipeline, so we don't want to return this as static just in case.